### PR TITLE
Snakecase clientVersion

### PR DESCRIPTION
**Pull request by linda-le1**
### What was wrong?
Inherited javascript syntax. This PR does not include a deprecation step, intended for v6.
Related to https://github.com/ethereum/web3.py/issues/2598

### How was it fixed?
Changed clientVersion to client_version where needed

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

#### Cute Animal Picture

shiba

### CODE REVIEW

File: tests/core/web3-module/test_clientVersion.py [hunk 1]
```diff
@@ -1,2 +0,0 @@
-def test_web3_clientVersion(w3):
-    assert w3.clientVersion.startswith("EthereumTester/")
```
Comment (kclowes) [inline tests/core/web3-module/test_clientVersion.py:2]: I may have just missed it, but it looks like this file got deleted instead of moved?

**Review by kclowes — COMMENTED** [review]
Nice work! 💪 🎉  Looks like there is a quick lint check that needs to be addressed, and a newsfragment is missing. I think most of us have `black` running on save, which _should_ take care of most lint errors. 

The newsfragment will be a new file called `newsfragments/2686.breaking.rst` and can just contain a sentence with what changed. 

Thank you!

File: tests/core/web3-module/test_clientVersion.py (hunk 1 above)
Comment (linda-le1) [reply tests/core/web3-module/test_clientVersion.py:2]: Thanks for catching this! Just forgot to push up my latest branch where I renamed this file.

**Comment by linda-le1** [comment]
@kclowes question on the linter: I ran black over the files in my PR and nothing seemed to change. Did I maybe miss something?

**Comment by kclowes** [comment]
Ah, sorry. `black` doesn't do anything with strings. You can either add a `#noqa: E501` at the end of the line, or split the string into two lines. Then we should be good to go!

**Review by kclowes — APPROVED** [review]
Looks great! Thank you!

**Pull request merged**